### ISSUE TITLE
More Listeners for Clipboard Events

### ISSUE THREAD

**Issue by marioawad** [hluk/CopyQ#1121]
I need to play sounds whenever clipboard events occur.

I successfully did this for when the clipboard gets new data and when the clipboard is cleared.

What I can't get working is to play another sound when you copy data to the clipboard and that same data is already there on the clipboard. 

Any way I can do that? 

Here's what I have so far after looking at "updateClipboardData" and "onClipboardChanged" (I wrote the play sound methods and they just play a sound file via PowerShell).

updateClipboardData = function() {
    if(isClipboard()) {
        updateTitle();
        showDataNotification();
        setClipboardData();
        if(hasData()) {
            playCaptureSound();
        }
        else {
            playEraseSound();
        }
    }
}

Thank you!

**Comment by hluk** [comment]
> What I can't get working is to play another sound when you copy data to the clipboard and that same data is already there on the clipboard.

Maybe check the latest stored item in the clipboard tab (something like `str(read(0)) == str(data(mimeText))`).

Or use `settings()` to store and retrieve latest content from configuration - you can just store checksums (`settings('lastClipboardTextChecksum', md5sum(text))` and later check `str(settings('lastClipboardTextChecksum')) == str(data(mimeText))`).

**Comment by hluk** [comment]
Oh, actually the problem probably is that if you copy the same text again, CopyQ won't call those callbacks - it assumes there is no reason to do that.

Hmm, I could add a separate callback function for this.

**Comment by marioawad** [comment]
Yes exactly. Both "updateClipboardData" and "onClipboardChanged" are never called when the same text is copied again.

Thank you for looking into this and for providing extra tips and info as well.

Cheers!
